katello/katello-change-hostname

```diff
@@ -245,20 +252,24 @@ if @foreman_proxy_content
 end
 
 # Get the hostname from your system
-old_hostname = get_hostname
+@old_hostname = get_hostname
 
 scenario_answers = YAML.load_file("/etc/foreman-installer/scenarios.d/#{@options[:scenario]}-answers.yaml")
 
 unless @foreman_proxy_content
   STDOUT.puts "\nUpdating default #{@proxy}"
-  default_capsule_id = hammer_cmd("capsule list | grep #{old_hostname} | awk '{ print $1 }'").chomp
-  # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
-  # For the same reason, we accept exit code 65 here.
-  hammer_cmd("capsule update --id #{default_capsule_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
+  proxy_id = get_default_proxy_id
+  if proxy_id
+    # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
+    # For the same reason, we accept exit code 65 here.
+    hammer_cmd("capsule update --id #{proxy_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
+  else
```

Comment:
This `else` branch will never be called as `get_default_proxy_id` will either return an id or error out itself.